yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -391,6 +391,50 @@ func (d *JsonQueryFunctionDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type UniqueNullsNotDistinctDetector struct {
+	query    string
+	detected bool
+}
+
+func NewUniqueNullsNotDistinctDetector(query string) *UniqueNullsNotDistinctDetector {
+	return &UniqueNullsNotDistinctDetector{
+		query: query,
+	}
+}
+
+// Detect checks if a unique constraint is defined which has nulls not distinct
+func (d *UniqueNullsNotDistinctDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_INDEX_STMT_NODE {
+		indexStmt, err := queryparser.ProtoAsIndexStmt(msg)
+		if err != nil {
+			return err
+		}
+
+		if indexStmt.Unique && indexStmt.NullsNotDistinct {
+			d.detected = true
+		}
```

Comment:
I don't see a need to convert here, and add a overhead.

`unique` and `nullsnotdistinct` are directly present in the protomsg as fields.
You can directly use `GetBoolField(msg, fieldName)` to get that.

Reply:
Hey @sanyamsinghal I am a little confused. So according to yesterday's discussion what should I do? Keep it as a struct implementation since it is inside the Detect() function?

Reply:
Yeah you can keep it as it is